### PR TITLE
Enable PMD gradle plugin

### PR DESCRIPTION
**Pull request by KarolJastrzebski**
This is part of quality assurance activity to standardise the code. Few simple rules (basic, braces, unnecessary and unusedcode) should be enough for a good start.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -113,6 +114,19 @@ sonarqube {
   }
 }
 
+pmd {
+  toolVersion = "5.8.1"
+  ignoreFailures = true
+  sourceSets = [sourceSets.main]
```
Comment (rkondratowicz) [inline build.gradle]: add `test` and `functionalTest`

File: build.gradle [hunk 2]
```diff
@@ -113,6 +114,19 @@ sonarqube {
   }
 }
 
+pmd {
```
Comment (timja-kainos) [inline build.gradle]: Is this instead of checkstyle?
i.e. this repo already has checkstyle on it

Comment (KarolJastrzebski) [reply build.gradle]: Rather a complementary check to be used within Sonar or Codacy.

File: build.gradle (hunk 1 above)
Comment (KarolJastrzebski) [reply build.gradle]: `pmd` for all sources is run during `check` task, and `pmdFunctionalTests` is also available.

Comment (rkondratowicz) [reply build.gradle]: This property is used to configure what source sets should be analyzed as part of the `check` and `build` tasks.

File: build.gradle (hunk 2 above)
Comment (timja-kainos) [reply build.gradle]: What benefit does it give? How much extra time does it take?

File: ruleset.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<ruleset name="Custom ruleset"
```
Comment (rkondratowicz) [inline ruleset.xml]: perhaps 'Draft Store' ruleset or something similar?

File: build.gradle (hunk 2 above)
Comment (KarolJastrzebski) [reply build.gradle]: It helps finding bad practices and takes as much as caring for quality does. ;)

Comment (timja-kainos) [reply build.gradle]: shouldn't sonar be used for this?

**Pull request merged**